tests/manage/storageclass/test_create_2_sc_with_1_pool_comp_rep2.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import pytest
+from ocs_ci.ocs.resources.pod import get_fio_rw_iops
+from ocs_ci.framework.testlib import ManageTest, tier1
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_external_mode,
+    skipif_ocs_version,
+)
+from ocs_ci.ocs.cluster import (
+    validate_compression,
+    validate_replica_data,
+)
+from ocs_ci.ocs.constants import CSI_RBD_STORAGECLASS_YAML, CEPHBLOCKPOOL
+from ocs_ci.utility import templating
+from ocs_ci.helpers.helpers import create_unique_resource_name
+from ocs_ci.ocs.defaults import ROOK_CLUSTER_NAMESPACE, RBD_PROVISIONER
+
+log = logging.getLogger(__name__)
+
+
+@tier1
+@skipif_external_mode
+@skipif_ocs_version("<4.6")
+@pytest.mark.polarion_id("OCS-2391")
+class TestMultipleScOnePoolRep2Comp(ManageTest):
+    """
+    Create new rbd pool with replica 2 and compression.
+    Attach it to 2 new storageclasses.
+    Create PVCs and PODs for each storageclass.
+    Run IO.
+    Delete PODs, PVDs, Storageclass and pool.
+
+    """
+
+    replica = 2
+
+    def test_multiple_sc_one_pool_rep2_comp(
+        self,
+        ceph_pool_factory,
+        storageclass_factory,
+        pvc_factory,
+        pod_factory,
+        secret_factory,
+    ):
+        """
+        This test function does below,
+        *. Creates 2 Storage Class with creating one rbd pool for both
+        *. Creates PVCs using new Storage Classes
+        *. Mount PVC to an app pod
+        *. Run IO on an app pod
+        *. Verify compression and replication
+
+        """
+
+        log.info("Creating new secret")
+        secret_obj = secret_factory(interface=CEPHBLOCKPOOL)
+
+        log.info("Creating new pool with replica2 and compression")
+        pool_obj = ceph_pool_factory(
+            interface=CEPHBLOCKPOOL,
+            replica=self.replica,
+            compression="aggressive",
+        )
+
+        # loading storageclass yaml and changing parameters
+        storageclass_dict = templating.load_yaml(CSI_RBD_STORAGECLASS_YAML)
```

Comment:
you can avoid changing this yaml here in the test by allowing `tests.conftest.storageclass_factory_fixture` to get the existing pool you created above

Reply:
I don't see in the factory and option to pass `interface_name`. You can pass `new_rbd_pool =False` but then it will take the default rbd pool as I understand from the code. What I can do if I'm right in the above is change the function to deal with existing rbd pool.

Reply:
I meant that we should add a new argument to the factory which is the pool name to use

Reply:
Great. Will do